src/main/java/com/mcsunnyside/coreprotecttnt/Main.java

```diff
@@ -74,6 +76,52 @@ public void run() {
         }.runTaskTimerAsynchronously(this, 0, 20 * 60);
     }
 
+    @EventHandler
+    public void onPlayerInteract(PlayerInteractEvent e) {
+        if (!getConfig().getBoolean("bed.log")) {
+            return;
+        }
+
+        Block bed = e.getClickedBlock();
+
+        if (e.getAction() != Action.RIGHT_CLICK_BLOCK ||
+                bed.getLocation().getWorld().getEnvironment() == World.Environment.NORMAL) {
+            return;
+        }
+
+        if (!(bed.getBlockData() instanceof Bed)) {
+            return;
+        }
+
+        Bed data = (Bed) bed.getBlockData();
+        Location location = bed.getLocation();
+        if(data.getPart() == Bed.Part.FOOT) {
+            location = location.add(data.getFacing().getDirection());
+        }
+
+        explosiveBlocks.put(location, e.getPlayer().getName());
+        api.logRemoval("#[Bed]" + e.getPlayer().getName(), location, bed.getType(), bed.getBlockData()); // head
+        api.logRemoval("#[Bed]" + e.getPlayer().getName(), location.subtract(data.getFacing().getDirection()),
+                bed.getType(), bed.getBlockData()); // foot
```

Comment:
You apply the subtraction to the same Location that is in explosiveBlocks, so the Location in explosiveBlocks becomes incorrect. To avoid this behavior, you can .clone() the location, and it will create a new independent instance of Location and won't affect anything else: `location.clone().subtract(...)`